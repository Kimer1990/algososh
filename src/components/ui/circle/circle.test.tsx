import { Circle } from "./circle";
import TestRenderer from "react-test-renderer";
import { ElementStates } from "../../../types/element-states";

describe("Circle", () => {
  it("renders correctly without letter", () => {
    const circle = TestRenderer.create(<Circle />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("renders correctly with letter", () => {
    const circle = TestRenderer.create(<Circle letter="L" />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("renders correctly with head", () => {
    const circle = TestRenderer.create(<Circle head="H" />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("renders correctly with React element in head", () => {
    const circle = TestRenderer.create(
      <Circle head={<span className="head">Head</span>} />
    ).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("renders correctly with tail", () => {
    const circle = TestRenderer.create(<Circle tail="T" />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("renders correctly with React element in tail", () => {
    const circle = TestRenderer.create(
      <Circle tail={<span className="tail">Tail</span>} tailType="element" />
    ).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("renders correctly with index", () => {
    const circle = TestRenderer.create(<Circle index={1} />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("renders correctly with isSmall prop", () => {
    const circle = TestRenderer.create(<Circle isSmall={true} />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("renders correctly in default state", () => {
    const circle = TestRenderer.create(
      <Circle state={ElementStates.Default} />
    ).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("renders correctly in changing state", () => {
    const circle = TestRenderer.create(
      <Circle state={ElementStates.Changing} />
    ).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("renders correctly in modified state", () => {
    const circle = TestRenderer.create(
      <Circle state={ElementStates.Modified} />
    ).toJSON();
    expect(circle).toMatchSnapshot();
  });
});
