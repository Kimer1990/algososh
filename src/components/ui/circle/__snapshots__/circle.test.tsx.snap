// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle renders correctly in changing state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly in default state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly in modified state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly with React element in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <span
      className="head"
    >
      Head
    </span>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly with React element in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <span
      className="tail"
    >
      Tail
    </span>
  </div>
</div>
`;

exports[`Circle renders correctly with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    H
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle renders correctly with isSmall prop 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly with letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      L
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    T
  </div>
</div>
`;

exports[`Circle renders correctly without letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
